Nitrogen

miladagreen opened the pull request:
- ✅ added nitrogen map
- ✅ upgraded bit map
- ✅ added required note on necessary inputs
- ✅ growth stage added
- ✅ *** NEEDS UPDATED BUILD WORKFLOW ***

Commit:
convert to vite initial stage

Commit:
convert to vite - fix styling issues

Commit:
linting

Commit:
only one covercrop selctable in sat mode

Commit:
only one covercrop selctable in sat mode

Commit:
only one covercrop selctable in sat mode

Commit:
added plant factor from growth stage api call

Commit:
added nitrogen map widget

Commit:
add required to fields

Commit:
fix nitrogen map

mikahpinegar committed:
revert dependencies

Commit:
added readme

Commit:
updated bit map to dynamica color range

Commit:
updated bit map to dynamica color range -

Commit:
updated bit

Commit:
updated bit

Commit:
Merge branch 'develop' into nitrogen

Commit:
separate nitrogen biomass maps

Commit:
added nitrogen fetch logic

Commit:
added nitrogen fetch logic

Commit:
useFetch axios

Commit:
snackbar progresive HLS status

Commit:
moved HLS fetch logic outside biomass page

Commit:
moved HLS fetch logi

Commit:
changes for build vite

Commit:
changed Dockerfile to adhere to vite and yarn

Commit:
added help badge when polygon is not drawn

Commit:
added help badge when polygon is not drawn

Commit:
fixed warning for using html and mui in help button of field name

Commit:
fixed warning for using html and mui in help button of field name

Commit:
fixed warning for using html and mui in help button of field name

Commit:
bipath for location page

Commit:
started satellite path one page design

Commit:
satpath one page

Commit:
aligning of widgets is fixed

Commit:
biomass trigger is fixed

Commit:
date picker moved to location tab

Commit:
date picker moved to location tab

Commit:
arranged widgets in a two columns grid

Commit:
arranged widgets in a two columns grid

Commit:
side nav jittering while scroll fixed

Commit:
loading example fix

rickhitchcock commented:
The User Sampled branch needs a cash crop planting date (found on the Cash Crop tab), and a cover crop termination date (found on the Cover Crop tab).

The program has unspecified planting and termination dates on the Location tab.  I think these can go away.

The planting date on the Location tab is actually the cover crop planting date, which isn't needed for the User Sampled branch.  (I'm also unsure why it's needed in the Satellite branch, but I need to study that some more.)

rickhitchcock commented:
The **User Sampled** branch creates a biomass map if a polygon is created.  I don't think this should happen – unless someone requested that feature(?)

rickhitchcock commented:
After choosing **User Sampled** and **Example: Grass**, the program crashes if you click **Cover Crop**.

rickhitchcock commented:
For the **User Sampled" path, **Biomass** Map** and **Nitrogen Map** should not be on the menu on the **Output** screen.

rickhitchcock commented:
This may be a question for the group, but I think that **Feedback** should always be visible rather than on the drop-down.  (This is the case for the [original program](https://covercrop-ncalc.org/), [the Selector](https://covercrop-selector.org/), [Vegspec](https://vegspec.org/), and [CC-Econ](https://covercrop-econ.org/)).

Also **Feedback** should be sent to the [Feedback API](https://github.com/precision-sustainable-ag/dst-feedback-api) rather than to [dst-ncalc/discussions](https://github.com/precision-sustainable-ag/dst-ncalc/discussions/15).

Commit:
fixed: changing the cover crop species doesn't update it in the summary widget

Commit:
fixed: changing species or growth stage doesn't update carbohydrates, holo-cellulose, or lignin in the summary widget

Commit:
fixed: nitrogen added to summary widget

Commit:
fixed: On the About screen, a space is needed after visit:
visithttps://precisionsustainableag.org/.

Commit:
removed about from drop-down menu in navbar

Commit:
fixed After choosing User Sampled and Example: Grass, the program crashes if you click Cover Crop.

Commit:
fixed: The User Sampled branch creates a biomass map if a polygon is created

Commit:
fixed: loading in an example after already defining a geometry should clear the old geometry

mikahpinegar committed:
npm i

mikahpinegar committed:
Update .eslintrc.cjs

YazidZidane committed:
remove yarn.lock and vscode settings

YazidZidane committed:
update satellite mode steppable

transfer to the user sampled mode selection

YazidZidane committed:
add shared components

added some comments

YazidZidane committed:
integrate header component

YazidZidane committed:
update header nav

YazidZidane committed:
update stepper

YazidZidane committed:
update stepper styling

remove previous navbar,  remove vscode settings

YazidZidane committed:
update comments, add api interval

YazidZidane committed:
update deps, replace ncalc map

YazidZidane committed:
update model api

YazidZidane committed:
update feedback api

YazidZidane committed:
move home to pages foldet

YazidZidane committed:
move Body to App

remove Body component

YazidZidane committed:
Revert "move home to pages foldet"

This reverts commit 182539b4e846fd000ee1523d447a0dc716e7ca0f.

YazidZidane committed:
move Location from shared to components

YazidZidane committed:
rename init to FieldDropdown

YazidZidane committed:
rename biomass to biomassdata

add comments

YazidZidane committed:
update field dropdown menu

YazidZidane committed:
add info modals for map, add comments

YazidZidane committed:
update fetch for biomass map

avoid redundant fetch

YazidZidane committed:
update for home page

move about to home page, update a redux value to state

YazidZidane committed:
update for soil page and hook api

ravighevariya10 committed:
Integrated the shared radio button

YazidZidane committed:
update for rest pages

add comment, update map, update format

ravighevariya10 committed:
Integrated the PSAForm

ravighevariya10 committed:
Integrated the PSAForm

YazidZidane committed:
Merge pull request #129 from precision-sustainable-ag/feature/PSAForm

Connect PSAForm shared component to NCalc #117

YazidZidane committed:
udpate dependencies

YazidZidane committed:
Merge branch 'feature/updateNcalc' into feature/radio-button

YazidZidane committed:
Merge pull request #128 from precision-sustainable-ag/feature/radio-button

Connect PSACheckbox and PSARadio shared component(s) to NCalc #118

YazidZidane committed:
add auth provider and login button

ravighevariya10 committed:
Integrated the PSADropdown component

ravighevariya10 committed:
reverted the autocomplete component

ravighevariya10 committed:
1. Added the comments back
2. Fixed the warning related to the PSADropdown item

YazidZidane committed:
Merge pull request #130 from precision-sustainable-ag/feature/PSADropdown

Connect PSADropdown shared component to NCalc #116

YazidZidane committed:
update location for saving polygon

YazidZidane committed:
update soil api for user history

not calling api if history state is imported

YazidZidane committed:
update deps

YazidZidane committed:
update for user sampled mode

YazidZidane committed:
remove large objects from user history

YazidZidane committed:
add data for satellite mode

YazidZidane committed:
create keys, authToken file

ravighevariya10 committed:
Integrated the shared loading indicator

YazidZidane committed:
add history api, loading history

YazidZidane committed:
update loading function

ravighevariya10 committed:
Merge branch 'feature/updateNcalc' into loading-indicator

YazidZidane committed:
Merge pull request #132 from precision-sustainable-ag/loading-indicator

Connect PSASpinner (for loading screen) shared component to NCalc #119

YazidZidane committed:
add create history function

remove biomass result from history

Diksha-Maurya committed:
Integrated PSATooltip

Diksha-Maurya committed:
redundant tooltip removed

YazidZidane committed:
prevent history with same name

ravighevariya10 committed:
Replaced the mui TextFields with the shared PSATextfield

Diksha-Maurya committed:
PSAModal implemented

Diksha-Maurya committed:
comments resolved

YazidZidane committed:
Merge pull request #133 from precision-sustainable-ag/feature/tooltip

Integrated PSATooltip

Diksha-Maurya committed:
Merge branch 'feature/updateNcalc' of https://github.com/precision-sustainable-ag/dst-ncalc into feature/modal

YazidZidane committed:
Merge pull request #135 from precision-sustainable-ag/PSATextfield

Connect PSATextField shared component to NCalc #110

YazidZidane committed:
add update history

hide localStorage lists when logged in

YazidZidane committed:
fix api reload on soil

update font on summary, add fetchtimeout for biomass api

YazidZidane committed:
update cover crop autocomplete

YazidZidane committed:
deprecate custom input component

replace with PSATextField

YazidZidane committed:
Merge branch 'feature/updateNcalc' into feature/userHistory

YazidZidane committed:
fix issues about dropdown

Diksha-Maurya committed:
Merge branch 'feature/updateNcalc' of https://github.com/precision-sustainable-ag/dst-ncalc into feature/modal

Diksha-Maurya committed:
previous model componenent removed

YazidZidane committed:
Merge pull request #136 from precision-sustainable-ag/feature/modal

Feature/modal

YazidZidane committed:
fix dropdown and soil

YazidZidane committed:
remove useless useeffect

YazidZidane committed:
remove clgs

YazidZidane committed:
remove clg

ravighevariya10 committed:
Integrated the shared PSAStepper

mikahpinegar committed:
Merge pull request #131 from precision-sustainable-ag/feature/updateNcalc

Updates for ncalc

YazidZidane committed:
Merge remote-tracking branch 'origin/nitrogen' into feature/userHistory

YazidZidane committed:
update deps and naming

nisarg210 committed:
Integrated PSAButton

YazidZidane committed:
Update package-lock.json

YazidZidane committed:
Merge branch 'feature/userHistory' into feature/Button

ravighevariya10 committed:
Integrated the shared Accordion

ravighevariya10 committed:
1. Implemented toggle functionality for Accordion expansion.
2. Added logic to verify the existence of field names.

YazidZidane committed:
Merge pull request #140 from precision-sustainable-ag/accordion

Connect PSAAccordion shared component to NCalc #114

nisarg210 committed:
Added style to Cancel button.

nisarg210 committed:
Merge branch 'feature/Button' of https://github.com/precision-sustainable-ag/dst-ncalc into feature/Button

YazidZidane committed:
Merge pull request #139 from precision-sustainable-ag/feature/Button

Integrated PSAButton

ravighevariya10 committed:
Replaced the Slider with PSASlider component

YazidZidane committed:
Merge branch 'feature/userHistory' into progress-stepper

YazidZidane committed:
update mobile stepper

YazidZidane committed:
Merge pull request #141 from precision-sustainable-ag/PSASlider

Connect PSASlider shared component to NCalc #122

YazidZidane committed:
update deps

YazidZidane committed:
Merge pull request #138 from precision-sustainable-ag/progress-stepper

Connect PSAStepper shared component to NCalc #125

Adamws33 committed:
Merge pull request #137 from precision-sustainable-ag/feature/userHistory

Add user history functions

ravighevariya10 committed:
1. Moved the field-name textfield to the home page
2. Added a description on the home page about the user history feature
3. Removed the accordion and combined it to the map box

YazidZidane committed:
Merge pull request #144 from precision-sustainable-ag/user-history

Update for user history user experience #142

YazidZidane committed:
move history list to front page

add alert for loading and saving history

YazidZidane committed:
remove unused files

YazidZidane committed:
update stylings in pages

YazidZidane committed:
fix stepper refresh problem

remove unused files

YazidZidane committed:
replace navigate buttons with bar

use a reuseable component instead

YazidZidane committed:
update for cover crop autocomplete

YazidZidane committed:
update front page styling

YazidZidane committed:
fix autocomplete and dropdown

remove clg

YazidZidane committed:
fix responsive issues

YazidZidane committed:
reset store when history name changed

YazidZidane committed:
create history component

YazidZidane committed:
Merge pull request #145 from precision-sustainable-ag/feature/updateUserHistory

Update user experience, page stylings

YazidZidane committed:
Update README.md

ravighevariya10 committed:
Merge pull request #146 from precision-sustainable-ag/feature/updateReadme

Update readme

YazidZidane committed:
add profile

YazidZidane committed:
update contrast colors

YazidZidane committed:
update highchart a11y

ravighevariya10 committed:
Merge pull request #147 from precision-sustainable-ag/feature/profile

Add Profile page

ravighevariya10 committed:
Merge pull request #148 from precision-sustainable-ag/feature/updateContrast

Update contrast colors

YazidZidane committed:
Merge branch 'nitrogen' into feature/keyboardNav

YazidZidane committed:
add skip content

YazidZidane committed:
update skipcontent style

YazidZidane committed:
add nav for slider

YazidZidane committed:
update key events for Help

YazidZidane committed:
fix annotation

ravighevariya10 committed:
Merge pull request #152 from precision-sustainable-ag/feature/keyboardNav

Keyboard navigation

YazidZidane committed:
update Help and annotation

ravighevariya10 committed:
Merge pull request #153 from precision-sustainable-ag/feature/annotation

Fixed annotation

rickhitchcock committed:
crash -> feedback

YazidZidane committed:
Update Dockerfile

YazidZidane committed:
update sticky stepper

YazidZidane committed:
update logic to disable next step

when data is not available

ravighevariya10 committed:
Merge pull request #172 from precision-sustainable-ag/feature/stickyStepper

Update sticky stepper

ravighevariya10 committed:
Merge pull request #173 from precision-sustainable-ag/feature/disableNext

update logic to disable next step

akshat22 committed:
replaced nitrogen map with PSAReduxMap

mikahpinegar committed:
Merge pull request #174 from precision-sustainable-ag/feature/map-merge

Use PSAReduxMap component
